docs/ql-documentation/learn-ql/cpp/introduce-libraries-cpp.rst

```diff
@@ -4,114 +4,686 @@ Introducing the C/C++ libraries
+| ``class`` *A* ``{`` ``friend`` ``void`` *f* ``(`` *int* ``);`` ``friend`` ``class`` *X* ``;`` ``};`` ``void`` *f* ``(`` *int x* ``)`` ``{`` … ``}`` ``class`` *X* ``{`` … ``};``                                                                                                                                                                                                                                                            | `FriendDecl <https://help.semmle.com/qldoc/cpp/semmle/code/cpp/FriendDecl.qll/type.FriendDecl$FriendDecl.html>`__                                                     |                                                                                                                                                                                                                                                                                   |
++---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``struct hostent *gethostbyaddr(const void *addr, socklen_t len, int type);``                                                                                                                                                                                                                                                                                                                                                               | `Gethostbyaddr <https://help.semmle.com/qldoc/cpp/semmle/code/cpp/models/implementations/Inet.qll/type.Inet$Gethostbyaddr.html>`__                                    |                                                                                                                                                                                                                                                                                   |
```

Comment:
Are things like `Gethostbyaddr` that are not imported by default within the scope of what we want to document here?

Reply:
@jbj raised the exact same point, and we can definitely remove them.  I thought that they may prove useful to you specifically, so I left them in for now.